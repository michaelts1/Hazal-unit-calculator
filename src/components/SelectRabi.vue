<script lang="ts">
import { defineComponent, PropType } from 'vue'
import type { Ruler } from '../types'

export default defineComponent({
	props: {
		defaultRuler: {
			type: String as PropType<Ruler>,
			default: 'גרח',
		},
	},

	emits: ['ruler-change'],

	data() {
		return {
			value: this.defaultRuler,
		}
	},
})
</script>

<template>
	<span>
		<label for="ruler-select">שיטת פסיקה:</label>
		<select
			id="ruler-select"
			v-model="value"
			class="form-select"
			@change="$emit('ruler-change', value)"
		>
			<option value="גרח">
				רב חיים נאה
			</option>
			<option value="חזון_איש">
				חזון איש
			</option>
		</select>
	</span>
</template>
