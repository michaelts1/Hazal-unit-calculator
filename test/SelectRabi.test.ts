import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import type { Ruler } from '../src/types'
import SelectRabi from '../src/components/SelectRabi.vue'

describe('Component SelectRabi', () => {
	it('exists', () => {
		expect(SelectRabi).toBeTruthy()
	})

	const wrapper = mount(SelectRabi, {
		props: {
			defaultRuler: 'גר"ח' as Ruler,
		},
	})

	it('has a select field with correct id and 2 option elements', () => {
		expect(wrapper.find('select').exists()).toBe(true)
		expect(wrapper.find('select').attributes()).toHaveProperty('id', 'ruler-select')
		expect(wrapper.findAll('option')).toHaveLength(2)
	})

	it('reacts to value changes', async () => {
		const select = wrapper.find('select')

		await select.setValue('חזון_איש')
		expect(wrapper.vm.value).toBe('חזון_איש')

		await select.setValue('גרח')
		expect(wrapper.vm.value).toBe('גרח')

		const rulerChangeEvents = wrapper.emitted('ruler-change') as unknown[]
		expect(rulerChangeEvents[0]).toEqual([ 'חזון_איש' ])
		expect(rulerChangeEvents[1]).toEqual([ 'גרח' ])

		const changeEvents = wrapper.emitted('change') as unknown[]
		expect(changeEvents).toHaveLength(2)
	})
})
