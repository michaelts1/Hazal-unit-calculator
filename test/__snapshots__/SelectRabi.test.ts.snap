// Vitest Snapshot v1

exports[`Component SelectUnit > matches snapshot 1`] = `
"<select>
  <option value=\\"טפח\\">טפח</option>
  <option value=\\"אצבע\\">אצבע</option>
</select>"
`;
